### PR TITLE
Extended CDS client to separate out fetch & download steps.

### PR DESCRIPTION
**Pull request by mahrsee1997**

### CODE REVIEW

**Review by alxmrs — APPROVED** [review]
A nit and a concern, but otherwise this LGTM.

File: weather_dl/download_pipeline/clients.py [hunk 1]
```diff
@@ -73,6 +74,44 @@ def license_url(self):
         pass
 
 
+class SplitCDSRequest(cds_api.Client):
+    """Extended CDS class that separates fetch and download stage."""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (mahrsee1997) [reply weather_dl/download_pipeline/clients.py]: Yes, we can do that. Thanks for pointing out.

File: weather_dl/download_pipeline/clients.py [hunk 2]
```diff
@@ -73,6 +74,44 @@ def license_url(self):
...
+
+    @retry_with_exponential_backoff
+    def _download(self, url, path: str, size: int) -> None:
+        self.info("Downloading %s to %s (%s)", url, path, cds_api.bytes_to_string(size))
+        start = time.time()
+
+        dir_path, file_name = os.path.split(path)
+        try:
+            subprocess.run(
+                ['aria2c', '-x', '16', '-s', '16', url, '-d', dir_path, '-o', file_name, '--allow-overwrite'],
```
[the remark this answers is not in the document]
Comment (mahrsee1997) [reply weather_dl/download_pipeline/clients.py]: While we have installed `aria2` in the [environment.yml](https://github.com/google/weather-tools/blob/main/environment.yml#L26) file, is it still necessary to implement a fallback mechanism ? 

Furthermore, I have relocated the transfer code to enhance its reusability.

[the remark this answers is not in the document]
Comment (alxmrs) [reply weather_dl/download_pipeline/clients.py]: Fair enough! Ignore my recommendation, if it's in the yaml file, that works for me.

**Pull request merged**